src/shogun/base/SGObject.cpp

```diff
@@ -800,3 +806,17 @@ bool CSGObject::type_erased_has(const BaseTag& _tag) const
 {
 	return self->has(_tag);
 }
+
+void CSGObject::subscribe_to_parameters(ParameterObserverInterface* obs)
```

Comment:
@vigsterkr I've added a simple `ParameterObserverInterface` class, as you suggested yesterday, to implement the behavior of `subscribe_to_parameters()`.